File: 0123/123.cpp
class Solution {
public:
    int bestTimeToBuyAndSellStockIii(TreeNode* root) {
        if(!root) return 0;
        return 1+max(bestTimeToBuyAndSellStockIii(root->left),bestTimeToBuyAndSellStockIii(root->right));
    }
};

File: 0123/123.cs
public class Solution {
    public int BestTimeToBuyAndSellStockIii(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(BestTimeToBuyAndSellStockIii(root.left), BestTimeToBuyAndSellStockIii(root.right));
    }
}